src/main/java/org/terasology/climateConditions/HypothermiaSystem.java

```diff
@@ -44,11 +67,26 @@ public void observeDangerZone(MovedEvent event, EntityRef player) {
         if (height < thresholdHeight && lastHeight >= thresholdHeight) {
             if (player.hasComponent(HypothermiaComponent.class)) {
                 player.removeComponent(HypothermiaComponent.class);
+                delayManager.cancelPeriodicAction(player, FROSTBITE_DAMAGE_ACTION_ID);
             }
         }
     }
 
     @ReceiveEvent(components = {HypothermiaComponent.class})
+    public void onHypothermia(OnAddedComponent event, EntityRef player) {
+        delayManager.addPeriodicAction(player, FROSTBITE_DAMAGE_ACTION_ID, initialDelay, healthDecreaseInterval);
+    }
+
+    @ReceiveEvent(components = {HypothermiaComponent.class})
+    public void onPeriodicFrostbite(PeriodicActionTriggeredEvent event, EntityRef player) {
+        if (event.getActionId().equals(FROSTBITE_DAMAGE_ACTION_ID)) {
+            applyFrostbiteDamagePlayer(player);
+            applyStunEffect(player, 1000);
+            playFrostbiteSound(player);
```

Comment:
👍 I really like how you structured this handler. Really concise, and one can understand what is happening when the periodic action triggers!